cluster-operator/src/main/java/io/strimzi/operator/cluster/model/Utils.java

```diff
@@ -0,0 +1,116 @@
+        try {
+            if (data.get(key) == null) { // value is not set in config map
+                return new Storage(Storage.StorageType.PERSISTENT_CLAIM);
+            }
+            return Storage.fromJson(new JsonObject(data.get(key)));
+        } catch (Exception e) {
+            throw new InvalidConfigMapException(key, " is corrupted");
+        }
+    }
+
+    public static String getNonemptyString(Map<String, String> data, String key, String defaultValue) {
+        if (data.get(key) == null) { // value is not set in config map -> will be used default value
+            return defaultValue;
+        }
+        if (data.get(key).length() == 0) {
+            throw new InvalidConfigMapException(key, " is empty");
+        }
+        return data.get(key);
+    }
+
+    public static String getJsonCorruptionBlame(DecodeException de, String config) {
+        String token = "";
+        token = de.getMessage();
+        if (token.contains("Illegal unquoted character")) {
+            return "JSON quotation";
+        }
+        if (token.contains("Failed to decode: Unexpected end-of-input")) {
+            return "JSON braces";
+        }
+        token = token.replace("Failed to decode: Unrecognized token '", "");
+        token = token.substring(0, token.indexOf("'"));
+
+        String[] entries = config.replace("{", "").replace("}", "")
+               .replace("\"", "").replace(" ", "")
+                .replace(",", "").split("\n");
+        int i;
+        String blame = "";
+        for (i = 0; i < entries.length; ++i) {
+            if (entries[i].split(":")[1].equals(token)) {
+                blame = entries[i].split(":")[0];
+                break;
+            }
+        }
+        return blame;
+    }
+
+    public static KafkaConnectConfiguration getKafkaConnectConfiguration(Map<String, String> data, String key) {
+        JsonObject jo = getJson(data, key);
+        return jo == null ? new KafkaConnectConfiguration(new JsonObject()) : new KafkaConnectConfiguration(jo);
+    }
+
+    public static ZookeeperConfiguration getZookeeperConfiguration(Map<String, String> data, String key) {
+        JsonObject jo = getJson(data, key);
+        return jo == null ? new ZookeeperConfiguration(new JsonObject()) : new ZookeeperConfiguration(jo);
+    }
+
+    public static KafkaConfiguration getKafkaConfiguration(Map<String, String> data, String key) {
+        JsonObject jo = getJson(data, key);
+        return jo == null ? new KafkaConfiguration(new JsonObject()) : new KafkaConfiguration(jo);
+    }
+
+    public static JsonObject getJson(Map<String, String> data, String key) {
+        String config = data.get(key);
+        if (config == null) {
+            return null;
+        }
+        try {
+            new JsonObject(config);
+        } catch (DecodeException de) {
+            throw new InvalidConfigMapException(getJsonCorruptionBlame(de, config), " corruptes JSON");
```

Comment:
Should it be `corrupted JSON`? (typo)

Reply:
Lets say ```key.name``` has invalid syntax. It breaks JSON and it says ```key.name corruptes JSON```. So it is ok, I guess.

Reply:
Ah, ok ... isn't it `corrupts` in that case? I don't know ... _corruptes_ sounds terribly German :-).